File: app/src/main/java/com/prey/managers/PreyWifiManager.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.managers;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;

import com.prey.PreyLogger;

public class PreyWifiManager {

    private WifiManager wifiMgr=null;
    private static PreyWifiManager _instance = null;
    private Context ctx=null;

    private PreyWifiManager(Context ctx) {
        this.ctx=ctx;
        wifiMgr = (WifiManager) ctx.getSystemService(Context.WIFI_SERVICE);
    }

    public static PreyWifiManager getInstance(Context ctx) {
        if (_instance == null)
            _instance = new PreyWifiManager(ctx);
        return _instance;
    }

    public boolean isWifiEnabled(){
        if(wifiMgr!=null)
            return wifiMgr.isWifiEnabled();
        else
            return false;
    }

    public  boolean isOnline() {
        ConnectivityManager cm = (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = cm.getActiveNetworkInfo();
        if (netInfo != null && netInfo.isConnectedOrConnecting()) {
            return true;
        }
        return false;
    }

    public WifiInfo getConnectionInfo(){
        try{
            if(wifiMgr!=null)
                return wifiMgr.getConnectionInfo();
        }catch(Exception e){
            PreyLogger.e("Error:"+e.getMessage(),e);
        }
        return null;
    }

    public String getSSID(){
        if (getConnectionInfo()!=null){
            String ssid= getConnectionInfo().getSSID();
            if(ssid!=null){
                ssid=ssid.replace("\"", "");
            }
            return ssid;
        }
        return null;
    }

}

File: app/src/main/java/com/prey/managers/PreyWifiManager.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.managers

import android.content.Context
import android.net.ConnectivityManager
import android.net.wifi.WifiInfo
import android.net.wifi.WifiManager

import com.prey.PreyLogger

class PreyWifiManager {

    fun isWifiEnabled(context: Context): Boolean {
        val wifiMgr = context.getSystemService(android.content.Context.WIFI_SERVICE) as WifiManager
        return wifiMgr.isWifiEnabled ?: false
    }

    fun isOnline(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return connectivityManager.activeNetworkInfo?.isConnectedOrConnecting ?: false
    }

    private fun getConnectionInfo(context: Context): WifiInfo? {
        try {
            val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
            return wifiManager.connectionInfo
        } catch (e: Exception) {
            PreyLogger.e("Error:${e.message}", e)
        }
        return null
    }

    fun getSSID(context: Context): String? {
        val connectionInfo = getConnectionInfo(context) ?: return null
        return connectionInfo.ssid?.replace("\"", "")
    }

    companion object {
        private var instance: PreyWifiManager? = null
        fun getInstance(): PreyWifiManager {
            return instance ?: PreyWifiManager().also { instance = it }
        }
    }
}